ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -432,6 +433,89 @@ applyPatchDisableCompileTimeOptimizations ghcFlavor =
           ("{-# OPTIONS_GHC -O0 #-}\n" ++)
           =<< readFile' file
 
+applyPatchGHCiInfoTable :: GhcFlavor -> IO ()
+applyPatchGHCiInfoTable ghcFlavor =
+  when(ghcFlavor == GhcMaster) $ do
+    writeFile infoTableHsc .
+      replace
+        (unlines newExecConItblBefore)
+        ("#if MIN_VERSION_rts(1,0,1)\n"   <>
+           unlines newExecConItblBefore   <>
+         "#else\n"                        <>
+           unlines newExecConItblAfter    <>
+         "#endif\n") .
+      replace
+        "fillExecBuffer :: CSize -> (Ptr a -> Ptr a -> IO ()) -> IO (Ptr a)\n"
+        ("#if MIN_VERSION_rts(1,0,1)\n"                                           <>
+           "fillExecBuffer :: CSize -> (Ptr a -> Ptr a -> IO ()) -> IO (Ptr a)\n" <>
+         "#endif\n") .
+      replace
+        "#error hi"
+        (unlines
+         [  "foreign import ccall unsafe \"allocateExec\""
+          , "  _allocateExec :: CUInt -> Ptr (Ptr a) -> IO (Ptr a)"
+          , ""
+          , "foreign import ccall unsafe \"flushExec\""
+          , "  _flushExec :: CUInt -> Ptr a -> IO ()" ])
+      =<< readFile' infoTableHsc
+  where
+    infoTableHsc = "libraries/ghci/GHCi/InfoTable.hsc"
+
+    newExecConItblBefore = [
```

Comment:
I somewhat wonder if we’re better off starting to work with patch files rather than dumb string replace but that’s definitely out of scope for this PR.